ocaml/sdk-gen/java/templates/Class.mustache

```diff
@@ -0,0 +1,177 @@
+     */
+    public static class Record implements Types.Record {
+        public String toString() {
+            StringWriter writer = new StringWriter();
+            PrintWriter print = new PrintWriter(writer);
+            {{#fields}}
+            print.printf("%1$20s: %2$s\n", "{{{name_camel}}}", this.{{{name_camel}}});
+            {{/fields}}
+            {{#is_event_class}}
+            print.printf("%1$20s: %2$s\n", "snapshot", this.snapshot);
+            {{/is_event_class}}
+            return writer.toString();
+        }
+
+        /**
+        * Convert a {{{class_name}}}.Record to a Map
+        */
+        public Map<String,Object> toMap() {
+            var map = new HashMap<String,Object>();
+            {{#fields}}
+            map.put("{{{name}}}", this.{{{name_camel}}} == null ? {{{default_value}}} : this.{{{name_camel}}});
+            {{/fields}}
+            return map;
+        }
+
+        {{#fields}}
+        /**
+         * {{{description}}}{{#publish_info}}
+         * {{{publish_info}}}{{/publish_info}}
+         */
+        @JsonProperty("{{{name}}}"){{#is_deprecated}}
+        @Deprecated(since = "{{{deprecated_release}}}"){{/is_deprecated}}
+        public {{{type}}} {{{name_camel}}};
+
+        {{/fields}}
+        {{#is_event_class}}
+            /**
+            * The record of the database object that was added, changed or deleted.
+            * The actual type will be VM.Record, VBD.Record, or similar.
+            */
+            public Object snapshot;
+        {{/is_event_class}}
+    }
+
+    {{/is_empty_class}}
+    {{#methods}}
+    /**
+    * {{{description}}}
+    * Minimum allowed role: {{{minimum_allowed_role}}}
+    * {{{publish_info}}}{{#is_deprecated}}
+    * @deprecated since {{{deprecated_release}}}{{/is_deprecated}}
+    *
+    * @param c The connection the call is made on{{#parameters}}
+    * @param {{{name_camel}}} {{{description}}} {{{publish_info}}}{{/parameters}}{{^returns_void}}
+    * @return {{#is_async}}Task{{/is_async}}{{^is_async}}{{{return_description}}}{{/is_async}}{{/returns_void}}
+    * @throws BadServerResponse Thrown if the response from the server contains an invalid status.
+    * @throws XenAPIException if the call failed.
+    * @throws IOException if an error occurs during a send or receive. This includes cases where a payload is invalid JSON.{{#errors}}
+    * @throws {{{name}}} {{{description}}}{{/errors}}
+    */{{#is_deprecated}}
+    @Deprecated(since = "{{{deprecated_release}}}"){{/is_deprecated}}
+    public{{#is_static}} static{{/is_static}} {{#is_async}}Task{{/is_async}}{{^is_async}}{{{return_type}}}{{/is_async}} {{name_camel}}{{#is_async}}Async{{/is_async}}(Connection c{{#parameters}}, {{{type}}} {{{name_camel}}}{{/parameters}}) throws
+    BadServerResponse,
+    XenAPIException,
+    IOException{{#errors}},
+    {{name}}{{/errors}} {
+        String methodCall = "{{#is_async}}Async.{{/is_async}}{{{object_name}}}.{{{name}}}";
```

Comment:
Not that it matters, but the generated sources will be easier to diff if you reverse the order of sync and async methods.